### PR TITLE
Core 954 setup service

### PR DESCRIPTION
**Pull request by MehdiZonjy**
a new kharon service that periodically scans pypi and artifactory for possible supply chain attacks. This service can be used to perform other artifactory  scans by defining new schedules and rake tasks

this pr introduces:
- bk pipeline
- kharon service definitions
- unit tests
- rake task for performing the scanning

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  - label: deploy to integration
+    branches: integration
+    agents:
+      queue: core-services-integration
+    plugins:
+      - "ssh://git@github.com/assuranceiq/styx#v1":
+          deployment-spec: kharon/integration.yml
+
+  - label: deploy to production
```
Comment (r-gill) [inline .buildkite/pipeline.yml:45]: Is there any gate to `production` deployment? Such as `integration` deployment must be successful?

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    branches: master
+    agents:
+      queue: core-services-production
```
Comment (r-gill) [inline .buildkite/pipeline.yml:48]: Do we need `concurrency_group` to ensure single deployment at a time?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (r-gill) [inline .travis.yml:1]: Delete this file, unless there's some Travis CI somewhere I don't know about 😜

File: foo [hunk 4]
```diff
@@ -0,0 +1,14 @@
+rake aborted!
```
Comment (r-gill) [inline foo:1]: Delete this file?

File: kharon/integration.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+spec:
+  name: artifactory-scanner
+  stage: integration
+  costCenter: engineering
+  owner: core
+  cpu: 1024
+  memory: 2048
+  environmentVariables:
+    ARTIFACTORY_PYPI_ENDPOINT: "https://assuranceiq.jfrog.io/artifactory/api/pypi/assurance-pypi-dev-local/simple"
+    ARTIFACTORY_USERNAME: "mzonjy"
```
Comment (r-gill) [inline kharon/integration.yml:12]: Can we not use a service account instead of tying to individual team member?

File: lib/scanner.rb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require_relative 'pypi'
+require_relative 'slack'
+require_relative 'misc'
+module Scanner
+  class Error < StandardError; end
```
Comment (r-gill) [inline lib/scanner.rb:7]: custom `Error` class is not being raised, do we need it?

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (MehdiZonjy) [reply .buildkite/pipeline.yml:45]: it shouldn't be necessary. integration deployment happens on. `integration` branch:
```

  - label: deploy to integration
    branches: integration

```

**Review by r-gill — APPROVED** [review]

**Pull request merged**